atomics/include/desul/atomics/Generic.hpp

```diff
@@ -568,7 +568,8 @@ template <typename T, class MemoryOrder, class MemoryScope>
 DESUL_INLINE_FUNCTION T atomic_load(T* const dest,
                                     MemoryOrder order,
                                     MemoryScope scope) {
-  return Impl::atomic_fetch_oper(Impl::LoadOper<T, const T>(), dest, T(), order, scope);
+  using T_nc = typename std::remove_const<T>::type;
+  return Impl::atomic_fetch_oper(Impl::LoadOper<T_nc, const T_nc>(), const_cast<T_nc*>(dest), T_nc(), order, scope);
```

Comment:
I would rather take `dest` as `T const * const` and return 
`Impl::atomic_fetch_oper(Impl::LoadOper<T, const T>(), const_cast<T*>(dest), T(), order, scope);`

Reply:
I don't know about that. What is the purpose of loading form a const array atomicly. Other than that you lied about it being const.

Reply:
I think I'm with @crtrott on this one.